From a99901d792bf27b7fdca0794a47afc3b527325ac Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Thu, 11 Feb 2016 13:02:36 +0200
Subject: [PATCH 13/38] ARM: dts: imx7d: cl-som-imx7: add support for NAND
 flash

Add support for NAND flash on GPMI interface.
The NAND option is mutually exlusive with eMMC option,
connected to USBHC3 (SD3) interface. This because GPMI
and USDHC3 share same pins.
The GPMI interface is disabled by default.
It supposed to be enabled via fdt command line utility
in the u-boot.

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 arch/arm/boot/dts/nxp/imx/imx7d-cl-som-imx7.dts | 46 +++++++++++++++++++++++++
 1 file changed, 46 insertions(+)

diff --git a/arch/arm/boot/dts/nxp/imx/imx7d-cl-som-imx7.dts b/arch/arm/boot/dts/nxp/imx/imx7d-cl-som-imx7.dts
index d439f37d4d66..53ee38abc5c3 100644
--- a/arch/arm/boot/dts/nxp/imx/imx7d-cl-som-imx7.dts
+++ b/arch/arm/boot/dts/nxp/imx/imx7d-cl-som-imx7.dts
@@ -18,6 +18,10 @@
 	model = "CompuLab CL-SOM-iMX7";
 	compatible = "compulab,cl-som-imx7", "fsl,imx7d";
 
+	aliases {
+		gpmi = &gpmi;
+	};
+
 	memory@80000000 {
 		device_type = "memory";
 		reg = <0x80000000 0x10000000>; /* 256 MB - minimal configuration */
@@ -184,6 +188,28 @@
 	};
 };
 
+&gpmi {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpmi_nand>;
+	status = "disabled";
+	nand-on-flash-bbt;
+
+	/* MTD partition table */
+	partition@0 {
+		label = "kernel";
+		reg = <0x00000000 0x00980000>;
+	};
+	partition@980000 {
+		label = "dtb";
+		reg = <0x00980000 0x00080000>;
+	};
+	partition@a00000 {
+		label = "rootfs";
+		reg = <0x00a00000 0x0>;
+	};
+};
+
+
 &i2c2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c2>;
@@ -404,6 +430,26 @@
 		>;
 	};
 
+	pinctrl_gpmi_nand: gpmi-nand {
+		fsl,pins = <
+			MX7D_PAD_SD3_CLK__NAND_CLE			0x71
+			MX7D_PAD_SD3_CMD__NAND_ALE			0x71
+			MX7D_PAD_SD3_DATA0__NAND_DATA00			0x71
+			MX7D_PAD_SD3_DATA1__NAND_DATA01			0x71
+			MX7D_PAD_SD3_DATA2__NAND_DATA02			0x71
+			MX7D_PAD_SD3_DATA3__NAND_DATA03			0x71
+			MX7D_PAD_SD3_DATA4__NAND_DATA04			0x71
+			MX7D_PAD_SD3_DATA5__NAND_DATA05			0x71
+			MX7D_PAD_SD3_DATA6__NAND_DATA06			0x71
+			MX7D_PAD_SD3_DATA7__NAND_DATA07			0x71
+			MX7D_PAD_SD3_STROBE__NAND_RE_B			0x71
+			MX7D_PAD_SD3_RESET_B__NAND_WE_B			0x71
+			MX7D_PAD_SAI1_TX_BCLK__NAND_CE0_B		0x71
+			MX7D_PAD_SAI1_TX_DATA__NAND_READY_B		0x74
+			MX7D_PAD_SAI1_MCLK__NAND_WP_B			0x71
+		>;
+	};
+
 	pinctrl_tsc2046: tsc2046grp {
 		fsl,pins = <
 			MX7D_PAD_GPIO1_IO14__GPIO1_IO14			0x34 /* PENDOWN */
-- 
2.20.1

